pkg/aspect/query/query.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package query
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+
+	"aspect.build/cli/pkg/aspecterrors"
+	"aspect.build/cli/pkg/bazel"
+	"aspect.build/cli/pkg/ioutils"
+)
+
+type PresetQuery struct {
+	Name        string
+	Description string
+	Query       string
+}
+
+type Query struct {
+	ioutils.Streams
+
+	bzl           bazel.Bazel
+	isInteractive bool
+
+	Presets   []*PresetQuery
+	ShowGraph bool
+}
+
+func New(streams ioutils.Streams, bzl bazel.Bazel, isInteractive bool) *Query {
+	return &Query{
+		Streams:       streams,
+		bzl:           bzl,
+		isInteractive: isInteractive,
+	}
+}
+
+func (q *Query) Run(_ *cobra.Command, args []string) error {
+	presets := make(map[string]*PresetQuery)
+	var names []string
+	for _, p := range q.Presets {
+		presets[p.Name] = p
+		names = append(names, fmt.Sprintf("%s: %s", p.Name, p.Description))
+	}
```

Comment:
1. I'd choose a more descriptive label for `names` like `presetNames`.
2. Since this is simply mapping `q.Presets` to another format, you know the final length of `names`, you better ask for the correct length since initialization and avoid memory copying on each `append`.
3. You would get weird displaying bugs for duplicated preset names. While we don't have a mechanism for allowing users to override query presets, my suggestion below should protect you.

```suggestion
	presetNames := make([]string, len(q.Presets))
	for i, p := range q.Presets {
		if _, exists := presets[p.Name]; exists {
			err := fmt.Errorf("duplicated preset query name %q", p.Name)
			return fmt.Errorf("failed to run 'aspect %s': %w", cmd.Use, err)
		}
		presets[p.Name] = p
		presetNames[i] = fmt.Sprintf("%s: %s", p.Name, p.Description)
	}
```